gufe/tokenization.py

```diff
@@ -309,6 +312,15 @@ def from_shallow_dict(cls, dct: Dict):
         """
         return from_dict(dct)
 
+    def copy_with_replacements(self, **replacements):
+        dct = self._to_dict()
+        if invalid := set(replacements) - set(dct):
+            raise TypeError(f"Invalid replacement keys: {invalid}. "
+                            f"Allowed keys are: {set(dct)}")
+
+        dct.update(replacements)
+        return self._from_dict(dct)
```

Comment:
document this

Reply:
Done in https://github.com/OpenFreeEnergy/gufe/pull/90/commits/cfa6122bacab27325d8e503554c1031b3d5e4da8. Take another look -- this PR is going to be blocking proper use of settings, so it needs to get in. (This particular method was just something else I had already tossed in when I started working on the more important stuff, and probably should have been a separate PR.)